packages/hierarchy-builder/src/test/Utils.ts

```diff
@@ -241,11 +250,5 @@ export async function waitFor(check: () => void, timeout?: number) {
 }
 
 export function createFakeQueryReader(rows: object[]): ECSqlQueryReader {
-  return {
-    async *[Symbol.asyncIterator](): AsyncIterableIterator<ECSqlQueryRow> {
-      for (const row of rows) {
-        yield row as ECSqlQueryRow;
-      }
-    },
-  };
+  return eachValueFrom(from(rows));
```

Comment:
Should our tests depend on `rxjs` + `rxjs-for-await`. Maybe it would be better to have simple functions `toAsyncIterableIterator` similar to `toArray`. I guess `eachValueFrom(from(rows))` as one liner looks clean but I feel it adds unnecessary diffuculty by converting `Array` -> `Observable` -> `AsyncIterableIterator`.

Reply:
made a change to stop using `rxjs` for this altogether